src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -110,10 +107,27 @@ public async Task<IEnumerable<IAccount>> GetAccountsAsync()
                         new AcquireTokenCommonParameters(), 
                         requestContext));
 
-                accounts = await cacheSessionManager.GetAccountsAsync(Authority).ConfigureAwait(false);
+                localAccounts = await cacheSessionManager.GetAccountsAsync(Authority).ConfigureAwait(false);
+            }
+
+            if (AppConfig.IsBrokerEnabled && ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth())
+            {
+                //Broker is available so accounts will be merged using home account ID with local accounts taking priority
+                var broker = ServiceBundle.PlatformProxy.CreateBroker(null);
+                brokerAccounts = await broker.GetAccountsAsync(AppConfig.ClientId).ConfigureAwait(false);
+
+                foreach(IAccount account in brokerAccounts)
+                {
+                    if (localAccounts.Where(x => x.HomeAccountId == account.HomeAccountId).Count() == 0)
```

Comment:
Use `Any(x => ...)` instead of Where + Count
Also, the string comparison needs to be done using String.Equals